### PR TITLE
Component declaration, properties, methods section

### PR DESCRIPTION
**Pull request by espinr**
Component declaration
- properties and methods
- built-in methods

(+@MichaelWangzitao )

***
<a href="https://pr-preview.s3.amazonaws.com/espinr/miniapp-components/pull/15.html" title="Last updated on May 31, 2023, 12:56 PM UTC (4183d38)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/miniapp-components/15/39b4830...espinr:4183d38.html" title="Last updated on May 31, 2023, 12:56 PM UTC (4183d38)">Diff</a>

### CODE REVIEW

**Comment by espinr** [comment]
I've merged all the branches (one per individual section) into one to have a full picture of the document: https://raw.githack.com/espinr/miniapp-components/all-new-sections/index.html#sec-declaration

File: index.html [hunk 1]
```diff
@@ -1,1642 +1,305 @@
...
-	  </table>
-  
-	</section>
+    <section id="sec-properties">
+      <h3>Component Properties and Methods</h3>
+      <p data-cite="MINIAPP-LIFECYCLE">
+        The scripting resources (e.g., `my-component.js`) use the `export` keyword to expose an object with the component's data properties and methods, making them available to the [=Document/View layer=].     
+      </p>
+      <p>
+        The component's data properties are usually defined by the key `props`, whose value is an object of key-value properties. The key in the object determines the property's name, and its value is an object that defines the <a href="https://tc39.es/ecma262/#sec-ecmascript-data-types-and-values">data type</a> (`type` member) and value (`value` member).
```
Comment (Sharonzd) [inline index.html]: Maybe we need to distinguish between external properties and internal data.
"properties" are the attributes passed from the parent component.
"data" is the internal attribute of the sub component.

for example:
```
// parent-component.html
<my-comp  text="hello">
```

```
// sub-component.html
<div>
  {{text}} {{color}}
</div>

// sub-component.js
export default {
  props: {
    text: {
      type: String,
      value: 'bar'
    }
  },
  color: 'red'
}
```

Comment (espinr) [reply index.html]: Thanks @Sharonzd , I've added also data attributes, along properties. 
It's in 4183d388f03e2081db785ca520d629353fca1591

Comment (espinr) [reply index.html]: Merging changes proposed for a clear understanding of the latest version of the document. Feel free to suggest more updates and changes opening issues. Thanks!

**Pull request merged**